### DISCUSSION TITLE
Miscellaneous question, improvement or feature request

### DISCUSSION THREAD

**Discussion by LckySndays** [blakeblackshear/frigate#14496] [Ask A Question]
My Environment : **frigate:40c6fda-tensorrt** with browser **firefox 131.0.3**

1. Is **"webui_url"** config is still valid? I have added this config for quite some times but cannot figure out where it is being applied. what my understanding is that it will create link or button to the URL specified in that config somewhere in the web UI 
```
# Optional: URL to visit the camera web UI directly from the system page. Might not be available on every camera.
    webui_url: ""
```
2. The option for genai **"use_snapshot"** I think currently only can be applied to camera level override, It seems I cannot apply that config on global manner for all camera as default
```
 genai: 
      # Optional: Use the object snapshot instead of thumbnails for description generation (default: shown below)
      use_snapshot: False
```
3. I think it is a good idea to add more button to Live View page for more quick access functionality. For example button for direct access to that camera motion tuner or debug menu setting page. sometimes it can be little bit exhausting when I see something not quite tuned right for my camera from Live View Page or even from Live Thumbnail and I need to manually use Setting Button -> Configuration Setting -> Motion Tuner/Debug -> Find/Locate that Camera especially in multiple CCTV environment like for example 20+ CCTV
![image](https://github.com/user-attachments/assets/e91115da-7c91-4b91-93c5-857371bf7c02)

4. The same thing for Explore -> Tracked Object Details, I think it is a good idea as well to add quick download button for the Thumbnail/Snapshot/Videos directly accessible from that modal menu
   ![image](https://github.com/user-attachments/assets/98139029-3e78-4206-9f95-d2ac27eecd3c)

5. Configuration to add custom button added to some UI that can be configure for certain custom functionality like manual trigger action or even simple custom URL would be nice to have. something like URL to Go2RTC Instance or OpenWebUI Instance for Ollama

**Comment by hawkeye217** [comment]
> My Environment : **frigate:40c6fda-tensorrt** with browser **firefox 131.0.3**
> 
>     1. Is **"webui_url"** config is still valid? I have added this config for quite some times but cannot figure out where it is being applied. what my understanding is that it will create link or button to the URL specified in that config somewhere in the web UI
> 

No, this is not currently implemented in the 0.14 UI anywhere.

> 
>     2. The option for genai **"use_snapshot"** I think currently only can be applied to camera level override, It seems I cannot apply that config on global manner for all camera as default
> 

Correct, this is a camera level setting only, as are several other genai config options. This is intentional.


>     3. I think it is a good idea to add more button to Live View page for more quick access functionality. For example button for direct access to that camera motion tuner or debug menu setting page. sometimes it can be little bit exhausting when I see something not quite tuned right for my camera from Live View Page or even from Live Thumbnail and I need to manually use Setting Button -> Configuration Setting -> Motion Tuner/Debug -> Find/Locate that Camera especially in multiple CCTV environment like for example 20+ CCTV
>        ![image](https://private-user-images.githubusercontent.com/13220698/378665812-e91115da-7c91-4b91-93c5-857371bf7c02.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1Njk0NTIsIm5iZiI6MTcyOTU2OTE1MiwicGF0aCI6Ii8xMzIyMDY5OC8zNzg2NjU4MTItZTkxMTE1ZGEtN2M5MS00YjkxLTkzYzUtODU3MzcxYmY3YzAyLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDAzNTIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWEzZjg0YTQwNjU3NTFkNTliYzg2MWYyZDlhYmUwYmEyZjBiNzJiZGY3YTE0YjFlNWEzMzBmZDUwNzRjODhkYzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.cAq-Sdz6QDPDMLNUBjEkrZokUknbZafv15PSl9u1JS8)
> 
>     4. The same thing for Explore -> Tracked Object Details, I think it is a good idea as well to add quick download button for the Thumbnail/Snapshot/Videos directly accessible from that modal menu
>        ![image](https://private-user-images.githubusercontent.com/13220698/378668602-98139029-3e78-4206-9f95-d2ac27eecd3c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjk1Njk0NTIsIm5iZiI6MTcyOTU2OTE1MiwicGF0aCI6Ii8xMzIyMDY5OC8zNzg2Njg2MDItOTgxMzkwMjktM2U3OC00MjA2LTlmOTUtZDJhYzI3ZWVjZDNjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDIyVDAzNTIzMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTNmMGVhM2NmYjU0NjRkMTFhNDVkYjZkNWM0NGE0OTJkMjhmNDQ4MmZjMzJkZjMzNjI4OTM5ODUxYmVhYmYzMGUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Q0lVZyq0gAcsEpRQ85P0Cuj7G7CqgXFJ0d6SkfXXoG8)

The download links are quickly accessible from the "3 dots" context menu on the thumbnail grid page.

> 
>     5. Configuration to add custom button added to some UI that can be configure for certain custom functionality like manual trigger action or even simple custom URL would be nice to have. something like URL to Go2RTC Instance or OpenWebUI Instance for Ollama

Thanks for your suggestions!

**Reply by LckySndays** [reply]
@hawkeye217 
```
The download links are quickly accessible from the "3 dots" context menu on the thumbnail grid page.
```
Thanks, I found it now, although I feel it still not quite easy accessible and can be further improved. since the default view is set to "Summary" by default and I need to change it to "Unfiltered Grid" first so that I can see the 3 dots that you mentioned

**Reply by hawkeye217** [reply]
> @hawkeye217 
> 
> ```
> 
> The download links are quickly accessible from the "3 dots" context menu on the thumbnail grid page.
> 
> ```
> 
> Thanks, I found it now, although I feel it still not quite easy accessible and can be further improved. since the default view is set to "Summary" by default and I need to change it to "Unfiltered Grid" first so that I can see the 3 dots that you mentioned

The summary display only shows the first 10 tracked objects per label. Clicking the small arrow at the end of the list will bring you to the full grid of tracked objects for that label.

**Reply by hawkeye217** [reply]
I've committed a change to allow you to right click/long tap on any of the thumbnails in the Explore summary view to bring up the same context menu available on the grid. https://github.com/blakeblackshear/frigate/pull/14505

**Reply by LckySndays** [reply]
@hawkeye217 
Nice, but I still feel like its better to add same functionality inside "Tracked Object Details" modal" no matter it is a button or right click.
Consider this scenario, when you see object of interest from the blurry thumbnail inside explore summary view, most of people will click it first to see more details for example to see the high resolution full image first before considering is it worth to download it, and when they do want to download it, there is no such quick interaction function in that modal to allow for direct download, instead they need to close the modal first and use the 3 dots or the newly added right click to be able to download.

**Reply by LckySndays** [reply]
@hawkeye217 
Same thing for Live View, when you see something might not be quite well tuned for example false motion trigger from the live view thumbnail, many people will click those thumbnail first to bring full/bigger live view of those camera, and when they decided to make some tweak, there are no such direct access to that camera's setting such as mask/zones, motion tuner or debug. Instead they need to click gear->Configuration Setting->Try to find those camera from there

**Reply by NickM-27** [reply]
You can just use the header to choose snapshot or video then download

**Reply by LckySndays** [reply]
@NickM-27 Not quite sure what you mean by header, can you elaborate more or attach some images

**Reply by NickM-27** [reply]
Looks like the buttons aren't shown on the snapshot tab like it is in the review detail dialog. That would be easy to add

**Reply by LckySndays** [reply]
@hawkeye217 will you consider this?
![image](https://github.com/user-attachments/assets/477aff6a-4723-40fd-bca2-f9eab89160ee)

1. Instead of "Debug" change it to "Camera Name" since those page is already on debug page/menu
2. To make it more clean and less cluttered, just name it WebUI {webui_url} and go2rtc {live->stream_name /links.html?src=XXX}

**Reply by hawkeye217** [reply]
Not every user maps port 1983 so a go2rtc dashboard link would likely confuse users here.

**Comment by baffles** [comment]
> Explore -> Tracked Object Details

Forgive my ignorance, but I've never seen this page before. How do you even get to it?

**Reply by hawkeye217** [reply]
This is a feature for the next version of Frigate. The OP is running a dev build.
